server/webapp/WEB-INF/rails.new/app/assets/new_javascripts/models/agents/agents.js

```diff
@@ -88,6 +107,47 @@ define(['mithril', 'lodash', 'string-plus',
       });
     };
 
+    this.sortBy = function (attrName, order) {
+
+      var sortByStatus = function () {
+        return this.sortByAgents(function (agent) {
+          var rank = {
+            "Pending":     1,
+            "LostContact": 2,
+            "Missing":     3,
+            "Building":    4,
+            "Cancelled":   5,
+            "Idle":        6,
+            "Disabled":    7
+          };
+          return rank[agent[attrName]()];
+        });
+      };
+
+      var sortByAlphaNumeric = function () {
+        return this.sortByAgents(function (agent) {
+          return agent[attrName]();
+        });
+      };
+
+      var sortAgents = _.isEqual(attrName, 'agentState') ? sortByStatus.bind(this) : sortByAlphaNumeric.bind(this);
```

Comment:
You're calling a function which returns a function. This can possibly be simplified by using just one level of indirection.